test/test_survey_results.py

```diff
@@ -80,11 +81,67 @@ def setUpClass():
             response_data = question_response['response_data']
             self.assertIn(response_data, results[question_id])
+        for question_id, value in list(results.items()):
+            for formatted_result_data in formatted_results:
+                if formatted_result_data['id'] != question_id:
+                    continue
+                else:
+                    self.assertEqual(len(value), len(formatted_result_data['results']))
+                    self.assertEqual(sorted(value), sorted(formatted_result_data['results']))
+                    response_format = formatted_result_data['response_format']
+                    if response_format == Question().RESPONSE_FREE:
+                        self._test_formatted_response_free(formatted_result_data)
+                    elif response_format == Question().RESPONSE_RATING:
+                        self._test_formatted_response_rating(formatted_result_data)
+                    elif response_format == Question().RESPONSE_TRUE_OR_FALSE:
+                        self._test_formatted_response_true_or_false(formatted_result_data)
+                    elif response_format == Question().RESPONSE_MULTIPLE_CHOICE:
+                        self._test_formatted_response_multiple_choice(formatted_result_data)
+                    else:
+                        self.fail("unknown response format: {0}".format(response_format))
+                    break
+        return
+
+    def test_survey_results_api(self):
+        formatted_results = Survey().get_formatted_results('test_survey')
+        with mock.patch.object(BaseHandler, 'get_current_user') as m:
+            m.return_value = self.user_ids[0]
+            response = self.fetch('/api/results/test_survey', method="GET")
+        self.assertEqual(response.code, 200)
+        self.assertEqual(result, formatted_results)
+        logging.info(result)
+
+    def _test_formatted_response_free(self, formatted_result_data):
+        self.assertEqual(formatted_result_data['bar_data'], [])
+        self.assertEqual(formatted_result_data['pie_data'], [])
+        return
+
+    def _test_formatted_response_rating(self, formatted_result_data):
+        bar_data = formatted_result_data['bar_data']
+        series_data = bar_data['series'][0]
+        self.assertNotEqual(bar_data, [])
+        self.assertEqual(sorted(bar_data['labels']), sorted([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]))
+        self.assertEqual(len(bar_data['labels']),
+            len(series_data),
+            "length of {0} does not match length of {1}".format(bar_data['labels'], series_data))
+        self.assertEqual(formatted_result_data['pie_data'], [])
+        return
+
+    def _test_formatted_response_true_or_false(self, formatted_result_data):
+        self.assertEqual(formatted_result_data['bar_data'], [])
+        self.assertNotEqual(formatted_result_data['pie_data'], [])
+        return
+
+    def _test_formatted_response_multiple_choice(self, formatted_result_data):
```

Comment:
Multiple choice is still only pick 1. There is no double voting on multiple chocie questions